controller/src/controller/models/domain/rsptag.py

```diff
@@ -106,6 +107,38 @@ class RSPImageType(Enum):
 ]
 
 
+def _calculate_date(tagdata: dict[str, str]) -> datetime.datetime | None:
```

Comment:
This should be a method rather than a standalone function. It's only useful in the context of the parser in the class.